ui/store/actions.ts

```diff
@@ -1120,9 +1124,39 @@ export function enableSnap(
 export function removeSnap(
   snapId: string,
 ): ThunkAction<void, MetaMaskReduxState, unknown, AnyAction> {
-  return async (dispatch: MetaMaskReduxDispatch) => {
-    await submitRequestToBackground('removeSnap', [snapId]);
-    await forceUpdateMetamaskState(dispatch);
+  return async (dispatch: MetaMaskReduxDispatch, state: MetaMaskReduxState) => {
+    dispatch(showLoadingIndication());
+    const { snaps } = state().metamask;
+
+    const isAccountsSnap =
+      snaps[snapId]?.manifest?.initialPermissions.snap_manageAccounts !==
```

Comment:
Please do not use the manifest to determine if a snap has a permission. Use the PermissionController.

Otherwise things will get very difficult for us when we expose more dynamic permissions.